utils/ssh.go

```diff
@@ -91,3 +91,31 @@ func ParsePrivateKey(pemText, passphrase string) (ssh.Signer, error) {
 	}
 	return nil, err
 }
+
+// NoDeadlineConn is a net.Conn that ignores deadlines
+type NoDeadlineConn struct {
+	net.Conn
+}
+
+func (c *NoDeadlineConn) SetDeadline(_ time.Time) error {
+	return nil // Ignore deadline setting
+}
+
+func (c *NoDeadlineConn) SetReadDeadline(_ time.Time) error {
+	return nil // Ignore read deadline setting
+}
+
+func (c *NoDeadlineConn) SetWriteDeadline(_ time.Time) error {
+	return nil // Ignore write deadline setting
+}
+
+// The crypto/ssh package does not support deadline methods.
+// Some db drivers(for Golang) such as of mongodb and oracle, directly call these deadline methods, which will cause an error.
+// While some db drivers(for Golang) they either have a check for deadline timeout and return an error if it is set or ignore if the timeout is not set, such as in mysql go driver. While in some driver they internally set the deadline methods such as in pgx driver.
+// So for drivers that directly call these deadline methods, we need to wrap the connection with a custom connection that ignores the deadline methods.
```

Comment:
Again a minor change to make this crisp and clear.

```suggestion
// crypto/ssh does not support deadlines.
//   - Required for: MongoDB & Oracle (call SetDeadline unconditionally).
//   - Not Required for: MySQL (conditional calls) & PostgreSQL (uses context).
```